### PR TITLE
feat: enable errorlint

### PR DESCRIPTION
**Pull request by gssbzn**
## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -29,6 +29,7 @@ linters:
     - depguard
     - dogsled
     - errcheck
+    - errorlint # errorlint is a linter for that can be used to find code that will cause problems with the error wrapping scheme introduced in Go 1.13. [fast: false, auto-fix: false]
```
Comment (andreaangiolillo) [inline .golangci.yml:32]:
```suggestion
    - errorlint # errorlint is a linter that can be used to find code that will cause problems with the error wrapping scheme introduced in Go 1.13. [fast: false, auto-fix: false]
```

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: .golangci.yml (hunk 1 above)
Comment (gssbzn) [reply .golangci.yml:32]: this is actually a copy/paste from `golangci-lint linters` do you want to open a PR there?

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Review by threebee — APPROVED** [review]
restamp

**Pull request merged**